### PR TITLE
add support for reading the extern alias  in 3.1.4xx

### PR DESCRIPTION
**Pull request by nkolev92**
Fixes https://github.com/dotnet/sdk/issues/11612

Porting https://github.com/dotnet/sdk/pull/11451 and https://github.com/dotnet/sdk/commit/7a05b62e54f1dc2e662757a761f1b05dac9a7ec3 to match the change made on the legacy package reference side which added support for extern alias in 16.7, see https://github.com/dotnet/NuGet.BuildTasks/commit/286d99efae21704989da40911380ee63c82e9c5a.

I imagine, I might have to port https://github.com/dotnet/sdk/commit/40256fc6a2a6006a32b653eebd9a351ff635a987
 to get the tests running under the correct conditions, but just wanted to get the PR going ASAP. 

cc @wli3 @dsplaisted @sfoslund

### CODE REVIEW

**Review by wli3 — APPROVED** [review]

**Comment by wli3** [comment]
@nkolev92 do you have QB approval. Once you get that you can merge

**Comment by nkolev92** [comment]
@wli3 
Do we need QB approval already? 

I have not done that, but I can certainly try to get it.

**Comment by wli3** [comment]
@nkolev92 my bad. It is M2 approval.

**Comment by wli3** [comment]
@dondrake could you approve this checkin to SDK?

**Comment by dondrake** [comment]
@wil3, sorry, wrong person...

**Comment by nkolev92** [comment]
welp :D 

I think this is the right one @ :) 

@dondr

**Comment by nkolev92** [comment]
@wli3 
I was hoping we can get this in P3. Thanks.

**Comment by dondr** [comment]
Approved

**Pull request merged**
